cpp/CMakeLists.txt

```diff
@@ -274,7 +274,7 @@ target_link_libraries(
   rapidsmpf
   PUBLIC rmm::rmm cudf::cudf CCCL::CCCL $<TARGET_NAME_IF_EXISTS:ucxx::ucxx>
          $<TARGET_NAME_IF_EXISTS:libcoro>
-  PRIVATE $<$<BOOL:${RAPIDSMPF_HAVE_NUMA}>:numa>
+  PRIVATE numa
```

Comment:
I don't know if we can do this, I don't think there's a suitable package in PyPI for NUMA, that was the primary reason this was made optional.